src/reporter/plugin-host.ts

```diff
@@ -0,0 +1,210 @@
+import chalk, { Chalk } from 'chalk';
+import indentString from 'indent-string';
+import {
+    identity,
+    escape as escapeHtml,
+    assignIn
+} from 'lodash';
+
+import moment from '../utils/moment-loader';
+import OS from 'os-family';
+import { wordWrap, removeTTYColors } from '../utils/string';
+import getViewportWidth from '../utils/get-viewport-width';
+import { DIFF_COLORS } from '../utils/diff/colors';
+import { Moment } from 'moment';
+import ReporterStreamController from '../runner/reporter-stream-controller';
+import { Stream } from 'stream';
+import TestRunErrorFormattableAdapter from '../errors/test-run/formattable-adapter';
+
+// NOTE: we should not expose internal state to
+// the plugin, to avoid accidental rewrites.
+// Therefore we use symbols to store them.
+
+/*global Symbol*/
+const stream          = Symbol();
+const wordWrapEnabled = Symbol();
+const indent          = Symbol();
+const errorDecorator  = Symbol();
+
+interface ReporterSymbols {
+    ok: string;
+    err: string;
+}
+
+export default class ReporterPluginHost {
+    public name: string;
+    public streamController: ReporterStreamController | null;
+    public chalk: Chalk;
+    public moment: Moment;
+    public viewportWidth: number;
+    public symbols: ReporterSymbols;
+
+    public constructor (plugin: any, outStream: Stream, name: string) {
+        this.name             = name;
+        this.streamController = null;
+
+        // Type 'Symbol' cannot be used as an index type.
+        // https://github.com/microsoft/TypeScript/issues/1863
```

Comment:
IMHO we do not need indexing here. `Type 'Symbol' cannot be used as an index type` means that you cannot declare dictionaries with Symbols as keys. Declaring symbol-based properties in interfaces works:
```
const sym: unique symbol = Symbol();

interface S {
    [sym]: number;
}
  
const s: S = {
    [sym]: 42
}
```